Purchases/Public/RCPurchases.h

```diff
@@ -264,6 +264,33 @@ NS_SWIFT_NAME(restoreTransactions(_:));
 */
 - (void)checkTrialOrIntroductoryPriceEligibility:(NSArray<NSString *> *)productIdentifiers
                                  completionBlock:(RCReceiveIntroEligibilityBlock)receiveEligibility;
+
+
+
+API_AVAILABLE(ios(12.2), macos(10.14.4))
+typedef void (^RCPaymentDiscountBlock)(SKPaymentDiscount * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.PaymentDiscountBlock);
+
+- (void)paymentDiscountForProductDiscount:(SKProductDiscount *)discount
+                                  product:(SKProduct *)product
+                               completion:(RCPaymentDiscountBlock)completion API_AVAILABLE(ios(12.2), macosx(10.14.4));
+
+
+/**
+ Purchase the passed `SKProduct`.
+ 
+ Call this method when a user has decided to purchase a product with an applied discount. Only call this in direct response to user input.
+ 
+ From here `Purchases` will handle the purchase with `StoreKit` and call the `RCPurchaseCompletedBlock`.
+ 
+ @note You do not need to finish the transaction yourself in the completion callback, Purchases will handle this for you.
+ 
+ @param product The `SKProduct` the user intends to purchase
+ 
+ @param discount The `SKPatmentDiscount` to apply to the purchase
+ */
+- (void)makePurchase:(SKProduct *)product
+        withDiscount:(SKPaymentDiscount * _Nullable)discount
+     completionBlock:(RCPurchaseCompletedBlock)completion NS_SWIFT_NAME(makePurchase(_:_:_:)) API_AVAILABLE(ios(12.2), macosx(10.14.4));
```

Comment:
Do you think maybe we should label the discount in swift?
`makePurchase(_:discount:_:)`?